### PR TITLE
CHE-5735 Provide consistent "new tab" behaviour

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
Displays a button within editor tabs clicking on which displays popup menu for adding new files.
Displays a button withing processes tabs clicking on which displays popup for opening terminal and ssh.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/5735

new editor

new terminal

#### Changelog

New button within editor tabs clicking on which displays popup menu for adding new files.
New button within processes tabs clicking on which displays popup for opening terminal and ssh.

#### Release Notes

New button within editor tabs clicking on which displays popup menu for adding new files.
New button within processes tabs clicking on which displays popup for opening terminal and ssh.

### CODE REVIEW

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/editor/EditorPartStackView.java [hunk 1]
```diff
@@ -48,28 +51,61 @@
...
     initWidget(UI_BINDER.createAndBindUi(this));
 
+    plusPanel.getElement().setInnerHTML(FontAwesome.PLUS);
+    plusPanel.addDomHandler(
+        new ClickHandler() {
+          @Override
+          public void onClick(ClickEvent clickEvent) {
+            if (addTabButtonClickListener != null) {
+              addTabButtonClickListener.onAddTabButtonClicked(
+                  getAbsoluteLeft(plusPanel.getElement()) + 15,
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/editor/EditorPartStackView.java]: 15 - could be const

**Review by ashumilova — APPROVED** [review]

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/editor/EditorPartStackView.java [hunk 2]
```diff
@@ -48,28 +51,63 @@
 
   interface PartStackUiBinder extends UiBinder<Widget, EditorPartStackView> {}
 
+  /** Listener to handle clicking on Add tab button. */
+  interface AddTabButtonClickListener {
+
+    /**
+     * Is called when Add tab button clicked.
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/editor/EditorPartStackView.java]: `Is called when Add tab button was clicked?`
OR
`Called when Add tab button has been clicked.`

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/processes/panel/ProcessesPanelView.java [hunk 3]
```diff
@@ -141,6 +141,12 @@ void addWidget(
      */
     void onCloseCommandOutputClick(ProcessTreeNode node);
 
+    /**
+     * Is called when user is going to close command tab.
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/processes/panel/ProcessesPanelView.java]: the same comment as [here](https://github.com/eclipse/che/pull/6156#discussion_r137267549)

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/processes/panel/ProcessesPanelView.java [hunk 4]
```diff
@@ -154,5 +160,13 @@ void addWidget(
 
     /** Is called when user has double clicked on console tab to maximize/restore the console. */
     void onToggleMaximizeConsole();
+
+    /**
+     * Is called when `Add Tab` button has been clicked.
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/processes/panel/ProcessesPanelView.java]: the same comment as [here](https://github.com/eclipse/che/pull/6156#discussion_r137267549)

**Review by azatsarynnyy — APPROVED** [review]

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/editor/AddEditorTabMenuFactory.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.part.editor;
+
+/** Factory for creating instances of AddEditorTabMenu */
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/editor/AddEditorTabMenuFactory.java]: Please, use links in javadocs for references to the classes. It allows to ensure that javadoc will be updated during some refactoring session and also links allow easy navigation in the generated javadoc.

 `{@link AddEditorTabMenu}`

**Review by vparfonov — COMMENTED** [review]
Some classes have imports with * 
- https://github.com/eclipse/che/pull/6156/files#diff-d8be8ad9ef6107cdf2682d15632ba026R16
- https://github.com/eclipse/che/pull/6156/files#diff-86ea0bf42bb9456c01ddd24588d2dd2bR21
Check please maybe some else

**Review by vparfonov — APPROVED** [review]

**Comment by vitaliy-guliy** [comment]
Everything's fixed.

**Pull request merged**